services/src/main/java/org/keycloak/broker/saml/mappers/UserAttributeMapper.java

```diff
@@ -99,61 +101,84 @@ public String getDisplayType() {
                 AttributeType attr = choice.getAttribute();
                 if (name != null && !name.equals(attr.getName())) continue;
                 if (friendly != null && !friendly.equals(attr.getFriendlyName())) continue;
 
-                List<Object> attributeValue = attr.getAttributeValue();
-                if (attributeValue == null || attributeValue.isEmpty()) return null;
-                return attributeValue.get(0).toString();
+                List<String> attributeValue = attr.getAttributeValue().stream()
+                        .filter(Objects::nonNull)
+                        .map(Object::toString)
+                        .collect(Collectors.toList());
+                if (name != null) {
+                    attributeToValues.put(name, attributeValue);
+                } else if (friendly != null) {
+                    attributeToValues.put(friendly, attributeValue);
+                }
             }
         }
-        return null;
+        return attributeToValues;
     }
 
     @Override
     public void updateBrokeredUser(KeycloakSession session, RealmModel realm, UserModel user, IdentityProviderMapperModel mapperModel, BrokeredIdentityContext context) {
         String attribute = mapperModel.getConfig().get(USER_ATTRIBUTE);
-        String value = getAttribute(mapperModel, context);
-        if (attribute.equalsIgnoreCase("email")) {
-            user.setEmail(value);
-        } else if (attribute.equalsIgnoreCase("firstName")) {
-            user.setFirstName(value);
-        } else if (attribute.equalsIgnoreCase("lastName")) {
-            user.setLastName(value);
+        Map<String, List<String>> attributeToValues = getAttribute(mapperModel, context);
+        String attributeName = getAttributeName(mapperModel);
+        if (attribute.equalsIgnoreCase(EMAIL)) {
+            setIfNotEmpty(user::setEmail, attributeToValues.get(attributeName));
+        } else if (attribute.equalsIgnoreCase(FIRST_NAME)) {
+            setIfNotEmpty(user::setFirstName, attributeToValues.get(attributeName));
+        } else if (attribute.equalsIgnoreCase(LAST_NAME)) {
+            setIfNotEmpty(user::setLastName, attributeToValues.get(attributeName));
         } else {
-            String current = user.getFirstAttribute(attribute);
-            if (value != null && !value.equals(current)) {
-                user.setSingleAttribute(attribute, value.toString());
-            } else if (value == null) {
+            List<String> currentAttributeValues = user.getAttributes().get(attribute);
+            List<String> newAttributeValues = attributeToValues.get(attributeName);
+            if (newAttributeValues != null
+                    && currentAttributeValues != null
+                    && !CollectionUtil.collectionEquals(newAttributeValues, currentAttributeValues)) {
+                user.setAttribute(attribute, newAttributeValues);
+            } else {
```

Comment:
In case when `CollectionUtil.collectionEquals(newAttributeValues, currentAttributeValues) == true`, the attribute is removed. Is it intended?